src/types/utility/branded.ts

```diff
@@ -0,0 +1,9 @@
+const LayerBrandTypeId: unique symbol = Symbol.for('@layerfi/Brand')
+
+type LayerBrand<in out ID extends string | symbol> = {
+  readonly [LayerBrandTypeId]: {
+    readonly [id in ID]: ID
+  }
+}
+
+export type LayerBranded<T, ID extends string | symbol> = T & LayerBrand<ID>
```

Comment:
I like this! Not sure though if dts can handle this properly. Typescript doesn't complain about this, but I'm getting an error with dts. Do get the same issue:

```
$ npm-dts generate -L debug
(...)
src/hooks/businessPersonnel/useBusinessPersonnel.ts(43,17): error TS4058: Return type of exported function has or is using name 'LayerBrandTypeId' from external module ".../layer-fi/components/src/types/utility/branded" but cannot be named.
(...)
```